.circleci/config.yml

```diff
@@ -16,4 +20,31 @@ jobs:
             - ./node_modules
       - run: npm run lint
       - run: npm test
-      # See https://circleci.com/docs/2.0/deployment-integrations/ for deploy examples
+      - persist_to_workspace:
+          root: ~/repo
+          paths: .
+
+  deploy:
+    <<: *common
+    steps:
+      - attach_workspace:
+          at: ~/repo
+      - run: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/repo/.npmrc
```

Comment:
Do we also need to add the `.npmrc` file?

Reply:
The `.npmrc` file has the token so we don't want to publish that to the repo. However, that does bring up another thing; npm will ignore everything in the `.gitignore` file but I think there may be a couple of other things we want to ignore like `.circleci` so I should add that. Thanks for asking!

Reply:
Ah gotcha. So where does the `.npmrc` file reside in order for this work? Sorry if these are basic questions!

Reply:
My specialty is just-in-time knowledge as well, didn't know CircleCI until a week ago and still don't feel like I know it. What's happening here is that we're checking out a "checkpointed" system image, pulling down the GitHub repo at master, and doing a clean install. If the tests pass we can do an `npm publish` but have to have credentials. Since that's secret and we don't want to let it be known, CircleCI stores the key securely and only makes it available in the environment. We have to echo that out to the `.npmrc` file before trying the publish.

Which brings up another point. At the moment the token configured in CircleCI is my account, one of the todos is to figure out how to get a token that isn't associated with a particular person, we probably should have an account something like "USDS-Publisher" on npm and use a token from that. But then we also need to store its password somewhere that appropriate people can log in if needed. I'll ask Maya about that too and see whtat USWDS is doing.

Reply:
great point!